IIWA World cleanup

naveenoid opened the pull request:
Inspired by the HSRb PR, the Diagram building methods are factored out and some API cleanup is done. 

+@liangfok -> feature reviewer

This is a necessary PR for my forthcoming controlled IIWA world demo so tagging this as high priority.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/4366)
<!-- Reviewable:end -->

liangfok commented:
I completed my first round of review. BTW, the CI error was a #4034 flake.

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEBleckIphewZsMlir:-KYEBleckIphewZsMlis:b-4h0rp3) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L21)):*
> ```cpp
> using systems::DiagramBuilder;
> using systems::RigidBodyPlant;
> using systems::ConstantVectorSource;
> ```

Alpha sort.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 59 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEC6PMq-g_N49CPpNY:-KYEC6PMq-g_N49CPpNZ:b-gdbx1l) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L59)):*
> ```cpp
>   builder->ExportInput(plant->get_input_port(0));
> 
>   drake::log()->debug("Plant Diagram built...");
> ```

BTW, for extra clarity, should this be "Plant and visualizer Diagram built..."?

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 83 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYECLVzvtulND7yx6IL:-KYECLVzvtulND7yx6IM:bfptyta) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L83)):*
> ```cpp
>               constant_value);
> 
>   // Cascades the constant source to the iiwa plant diagram. This effectively
> ```

BTW, should "plant" be "plant and visualizer"?

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 107 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYECg09j07A6MC_t6j6:-KYECg09j07A6MC_t6j7:b-foviv7) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L107)):*
> ```cpp
>       dynamic_cast<const RigidBodyPlant<double>*>(
>           plant_and_visualizer_subsystems.at(0));
> ```

BTW: to be consistent with the previous `DRAKE_DEMAND`, this empty line should not be here.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 19 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYECwK_SLP4NE3L-gUN:-KYECwK_SLP4NE3L-gUO:bp0115w) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L19)):*
> ```objc
> /// Builds a diagram composed of a `RigidBodyPlant` and `DrakeVisualizer` from
> /// a `RigidBodyTree` and returns it. The plant's output port zero is connected
> /// to
> ```

BTW: Odd line break.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 37 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYED2TfLg5l98QBTsq4:-KYED2TfLg5l98QBTsq5:bp0115w) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L37)):*
> ```objc
> /// Sets the zero configuration in a `RigidBodyPlant` underlying an assembled
> /// diagram by accessing the context from the `Simulator`. Assumes that the
> /// diagram
> ```

BTW: Odd line break.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 38 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDLDo6kGMLdGI6WH6:-KYEDLDo6kGMLdGI6WH7:b-gypmvc) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L38)):*
> ```objc
> /// diagram by accessing the context from the `Simulator`. Assumes that the
> /// diagram
> /// was built by one of the factory methods above.
> ```

This method only works with the `Diagram` returned by `BuildConstantSourceToPlantDiagram()` right?

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.cc, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEFHmRc90Ru6lUJotF:-KYEFHmSNSell9WICfNi:bp6z4ke) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.cc#L106)):*
> ```cpp
> template <typename T>
> void WorldSimTreeBuilder<T>::StoreModel(const std::string& model_name,
>                                         const std::string& model_path) {
> ```

For extra safety, add a `DRAKE_DEMAND(model_map_.find(model_name) == model_map_.end())` statement here.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDq5pgJFXcfIso0a8:-KYEDq5pgJFXcfIso0a9:b-ca2qfs) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L13)):*
> ```objc
> // TODO(naveenoid): Consider a common location for this class.
> 
> /// A helper class to construct Robot world trees from model (urdf/sdf) files ;
> ```

Extra space before ";". Actually, this semi-colon should be a period.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEE5FcWVwqre27oB5a:-KYEE5FcWVwqre27oB5b:bjcbvk7) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L13)):*
> ```objc
> // TODO(naveenoid): Consider a common location for this class.
> 
> /// A helper class to construct Robot world trees from model (urdf/sdf) files ;
> ```

"Robot world trees" should be "robot world RigidBodyTree objects"

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEEC_LvXdIVIleTDiH:-KYEEC_LvXdIVIleTDiI:b-iivu4p) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L13)):*
> ```objc
> // TODO(naveenoid): Consider a common location for this class.
> 
> /// A helper class to construct Robot world trees from model (urdf/sdf) files ;
> ```

"urdf/sdf" should be "i.e., URDF/SDF" since they are acronyms.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDoJgzl3-lM25MUjb:-KYEDoJgzl3-lM25MUjc:bp0115w) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L14)):*
> ```objc
> 
> /// A helper class to construct Robot world trees from model (urdf/sdf) files ;
> /// Models
> ```

BTW: Odd line break.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEEOtyLXi0SJG5CqHf:-KYEEOtyLXi0SJG5CqHg:b-ffuh1h) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L15)):*
> ```objc
> /// A helper class to construct Robot world trees from model (urdf/sdf) files ;
> /// Models
> /// (Robots, objects for manipulation etc.) can be stored and added to the tree
> ```

"(Robots, objects for manipulation etc.)" should be "(e.g., robots, objects for manipulation, etc.)".

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDvivU7bQUjgnvPsA:-KYEDvivU7bQUjgnvPsB:bp0115w) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L16)):*
> ```objc
> /// Models
> /// (Robots, objects for manipulation etc.) can be stored and added to the tree
> /// to be
> ```

BTW: Odd line break.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEEviAlA68h9FmxwWs:-KYEEviAlA68h9FmxwWt:b-h4tp6t) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L32)):*
> ```objc
> 
>   /// Adds a fixed model instance specified by its name, @p model_name, to the
>   /// `RigidBodyTree` at the pose specified by position @p xyz and orientation
> ```

BTW, consider replacing "at" with "being built at".

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEF1D_Set7gcUKk3Q_:-KYEF1D_Set7gcUKk3Qa:b-h4tp6t) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L42)):*
> ```objc
> 
>   /// Adds a floating model instance specified by its name, @p model_name, to
>   /// the `RigidBodyTree` at the pose specified by position @p xyz and
> ```

BTW, consider replacing "at" with "being built at".

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 52 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEF24Po0dGm-ikw3o8:-KYEF24Po0dGm-ikw3o9:b-h4tp6t) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L52)):*
> ```objc
> 
>   /// Adds a model instance specified by its model name, @p model_name, to
>   /// the `RigidBodyTree` at a pose specified by position @p xyz and
> ```

BTW, consider replacing "at" with "being built at".

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 75 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEF8pm7yjfJ8SLP26o:-KYEF8pm7yjfJ8SLP26p:box8c95) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L75)):*
> ```objc
>   /// @p model_name coupled with URDF/SDF paths in @p model_path. Instances
>   /// of these models can then be added to the world via the various
>   /// `AddFoo()` methods provided by this class.
> ```

Mention in this paragraph that `@p model_name` is user-selectable but must be unique among all of the models that are stored.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 82 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEFWGhzvufJWN4xz0q:-KYEFWGhzvufJWN4xz0r:bfpex0y) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L82)):*
> ```objc
>   void StoreModel(const std::string& model_name, const std::string& model_path);
> 
>   /// Gets a pointer to the `RigidBodyTree` that was built.
> ```

Add "unique" before "pointer". Also, might as well also add a statement saying:

1. This method can only be called if it was not previously called.
2. Ownership of the manufactured RigidBodyTree is transferred to the calling code.
3. The instance of this class should be discarded after this method is called.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 89 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEG1fun-fCmtv3HhdX:-KYEG1fun-fCmtv3HhdY:blx3lvr) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L89)):*
> ```objc
>   }
> 
>   // TODO(naveenoid): create a method to reset the tree.
> ```

Instead of allowing the user to reset the tree, consider requiring that users simply discard and instantiate a new `WorldSimTreeBuilder`. I believe this "disposable" semantics matches that of `DiagramBuilder`.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 96 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEGGV1qsM29nfn8W_u:-KYEGGV1qsM29nfn8W_v:bgplisi) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L96)):*
> ```objc
>   bool built_{false};
> 
>   // Maps between models and their names. Instances of these models can be
> ```

For extra clarity, replace "Maps between models and their names." with "Maps between models (values in the map) and their user-supplied names (keys in the map)."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYEGUW50DgNJo25umlG:bcnkdgn)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Ready for the next round..Thanks and PTAL!

---

Review status: all files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEBleckIphewZsMlir:-KYEIFW1ZCingKMhigh3:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L21)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Alpha sort.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 59 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEC6PMq-g_N49CPpNY:-KYEIXgWzoYMk1yrmhbm:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L59)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, for extra clarity, should this be "Plant and visualizer Diagram built..."?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 83 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYECLVzvtulND7yx6IL:-KYEIeQjcv4H61fckGa4:b-v5pos7) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L83)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, should "plant" be "plant and visualizer"?
</blockquote></details>

Done. Also removed the "iiwa".

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 107 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYECg09j07A6MC_t6j6:-KYEIl-77QwdbSw2lHXm:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L107)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: to be consistent with the previous `DRAKE_DEMAND`, this empty line should not be here.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 19 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYECwK_SLP4NE3L-gUN:-KYEIsEpwzQcZ_Xg4THm:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L19)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: Odd line break.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 37 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYED2TfLg5l98QBTsq4:-KYEIw0rJUkWPEIiO-VY:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L37)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: Odd line break.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 38 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDLDo6kGMLdGI6WH6:-KYEKJB-LgqIkED-_qDC:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L38)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

This method only works with the `Diagram` returned by `BuildConstantSourceToPlantDiagram()` right?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.cc, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEFHmRc90Ru6lUJotF:-KYEKXa3HX1-kNK4Pn4D:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.cc#L106)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

For extra safety, add a `DRAKE_DEMAND(model_map_.find(model_name) == model_map_.end())` statement here.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDq5pgJFXcfIso0a8:-KYEKr_uVCORcJYCi9_s:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Extra space before ";". Actually, this semi-colon should be a period.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEE5FcWVwqre27oB5a:-KYEL1ZXQfWTcez69tPb:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

"Robot world trees" should be "robot world RigidBodyTree objects"
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEEC_LvXdIVIleTDiH:-KYEL52Y7Dw_-OwGho7y:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

"urdf/sdf" should be "i.e., URDF/SDF" since they are acronyms.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDoJgzl3-lM25MUjb:-KYEL6_EKfNJcA7VN3mP:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L14)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: Odd line break.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEEOtyLXi0SJG5CqHf:-KYELJF8RirjR17tRSJl:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L15)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

"(Robots, objects for manipulation etc.)" should be "(e.g., robots, objects for manipulation, etc.)".
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEDvivU7bQUjgnvPsA:-KYELJuYxA0xsr0Fbm9c:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L16)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: Odd line break.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEEviAlA68h9FmxwWs:-KYELk2y0sNVYN_oGEAy:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L32)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing "at" with "being built at".
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEF1D_Set7gcUKk3Q_:-KYELl0HL-y8TyVPqjIQ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L42)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing "at" with "being built at".
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 52 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEF24Po0dGm-ikw3o8:-KYELlin9ZOLhvCL99Pq:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L52)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing "at" with "being built at".
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 75 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEF8pm7yjfJ8SLP26o:-KYELzdQJ24g9sLqNuQ_:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L75)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Mention in this paragraph that `@p model_name` is user-selectable but must be unique among all of the models that are stored.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 82 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEFWGhzvufJWN4xz0q:-KYEMR8E3u4lhE73sT_9:bh8qsz6) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L82)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Add "unique" before "pointer". Also, might as well also add a statement saying:

1. This method can only be called if it was not previously called.
2. Ownership of the manufactured RigidBodyTree is transferred to the calling code.
3. The instance of this class should be discarded after this method is called.
</blockquote></details>

Done. Didnt use the itemized list though.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 89 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEG1fun-fCmtv3HhdX:-KYEMarMJo2sWMEtB2BO:b-jm027v) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L89)):*
> users simply discard and instantiate a new WorldSimTreeBuilder
Removed this TODO in the light of the note for the method above.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 96 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEGGV1qsM29nfn8W_u:-KYEMwn3BPscd34tE_zB:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L96)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

For extra clarity, replace "Maps between models and their names." with "Maps between models (values in the map) and their user-supplied names (keys in the map)."
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYEMyXlEek4M6yg8w0I:bj94517)*
<!-- Sent from Reviewable.io -->

liangfok commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> once a few minor BTW comments are addressed.

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 36 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEUEC9woE18K096dDG:-KYEUECAyRXpb-UZ4doM:b8aqawx) ([raw file](https://github.com/robotlocomotion/drake/blob/dbc96116a106ef0a20cc830cc24c3b91ca8e561b/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L36)):*
> ```objc
> /// Sets the zero configuration in a `RigidBodyPlant` underlying an assembled
> /// diagram by accessing the context from the `Simulator`.
> /// The argument @p diagram is to be generated using the
> ```

BTW, replace "The argument" with "Parameter".

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 37 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEUIYgZ4UZt7QyZQKO:-KYEUIYgZ4UZt7QyZQKP:b-rkjfc) ([raw file](https://github.com/robotlocomotion/drake/blob/dbc96116a106ef0a20cc830cc24c3b91ca8e561b/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L37)):*
> ```objc
> /// diagram by accessing the context from the `Simulator`.
> /// The argument @p diagram is to be generated using the
> /// `BuildConstantSourceToPlantDiagram()` method.
> ```

BTW, you can remove the back ticks and also the word "method" since Doxygen should automatically make it a hyperlink to the method's description.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 89 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEG1fun-fCmtv3HhdX:-KYEVGJCyq_pbkEpSpGO:b-9jlhsn) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L89)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

> users simply discard and instantiate a new WorldSimTreeBuilder
Removed this TODO in the light of the note for the method above.
</blockquote></details>

BTW, It looks like the TODO is still there.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYEVPq7OlqbtNWZLZej:b-yb0byi)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Thanks! Will resolve them. 
@sherm1 -> platform review.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYEVmhYi29Uc-OTGnSQ:b-csay89)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
(I'm out of reviewing gas for today.) +@david-german-tri for platform review.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYHnjGj46aXVjkFegZl:booisz7)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 36 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEUEC9woE18K096dDG:-KYIMki37Okd3NLvLB-u:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/dbc96116a106ef0a20cc830cc24c3b91ca8e561b/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L36)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, replace "The argument" with "Parameter".
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 37 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEUIYgZ4UZt7QyZQKO:-KYIMnt45iWIYNRq7PrT:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/dbc96116a106ef0a20cc830cc24c3b91ca8e561b/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L37)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, you can remove the back ticks and also the word "method" since Doxygen should automatically make it a hyperlink to the method's description.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 89 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYEG1fun-fCmtv3HhdX:-KYIMQNO6VNb-_MX9XGT:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ce0ba9bd5687b647ccfb62be3fe4618e36e5a8f9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L89)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, It looks like the TODO is still there.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
+(status: curate commits before merging) 

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYKXE2D-lOg6CIDz1JL:b-s794vx)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
CI is red and there are merge conflicts.  Due to timezones, I'll take an initial pass through the review anyway and leave any major comments I have, but I'll need to do another pass once the breakage is resolved.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYKXJjh2WsSXmckyg-q:b-bptwtg)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: 3 of 7 files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKs2Cg71XyOvVgHvH-:-KYNO9Gp8sr1zr9KW508:b-wh0uih) (no related file):*
> #4357
The modified iiwa_world_demo is currently accessing all of the underlying methods. It is included as a unit test. I can add a custom unit test of these 2 classes too but I prefer doing them as a follow up PR. Please let me know what is appropriate.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 91 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYK_-DKEFdLBk5MaoRC:-KYNBi-q9lVmqkFUWZ7B:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L91)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

Now that we have `System<T>::SetDefaultState` in #4357, can you just use that?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 107 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKrjacBrfIu3bVtDSp:-KYNmD40AjhSQeN6mq05:b-n9otrh) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L107)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

If you do keep this method, this code is an indication that you should really be using Diagram subclasses with named accessors, and helper functions for rummaging around in the context.  There's some related discussion on #4365.
</blockquote></details>

Done. Removed this method since its no longer needed. 

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 21 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKrKsg4STIlYbaWSZh:-KYODUhI0kRd_wKdH2qZ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L21)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

Consider making this, and the method below, subclasses of Diagram, as in https://github.com/RobotLocomotion/drake/blob/master/drake/systems/framework/test/diagram_test.cc#L27.  Alternatively, consider passing in a `DiagramBuilder`.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 32 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKqzL9EBhb4CBXKukL:-KYNtzdT0o3Zhoi9b2wB:b-sae872) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L32)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

did you mean `plant_diagram`?
</blockquote></details>

Done. Removed the need 

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 35 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYK_Fnl4F3pU7nSi02j:-KYNtz2kBsuUswo7_Qn2:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L35)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

If you do still need this (and as noted above I suspect you don't) please pass the context and not the entire simulator.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 87 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKqhNk5Uja1qJnxG-H:-KYNBZIO65-RLKsKl8to:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L87)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW consider just calling this `Build`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 2 of 5 files at r4.
Review status: 5 of 7 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 21 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKrKsg4STIlYbaWSZh:-KYOOwOs4M9YpcsSOx1G:b-dg1m8z) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L21)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I'm not sure I agree with the creation of a `VisualizedPlant` that is a child class of `Diagram` since it will not reduce the degree of nested `Diagram` objects in the ultimate `Diagram`. In the longer term, I believe we're going to create factory  methods that assembles the full `Diagram` to be simulated without nesting other `Diagram` objects by operating on a `DiagramBuilder` object. Thoughts?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: 5 of 7 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 21 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKrKsg4STIlYbaWSZh:-KYOPgMU7ef_V1Ei-Tjy:bbnrnea) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L21)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

I'm not sure I agree with the creation of a `VisualizedPlant` that is a child class of `Diagram` since it will not reduce the degree of nested `Diagram` objects in the ultimate `Diagram`. In the longer term, I believe we're going to create factory  methods that assembles the full `Diagram` to be simulated without nesting other `Diagram` objects by operating on a `DiagramBuilder` object. Thoughts?
</blockquote></details>

I personally have no thoughts either way on this. With the new methods of #4357  I believe in either case it isnt too complex to build and operate. I will stick to what you guys agree upon.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

liangfok commented:
*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 21 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKrKsg4STIlYbaWSZh:-KYOX4nv5-UHFRfxd3FR:b-od5a4k) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L21)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

I personally have no thoughts either way on this. With the new methods of #4357  I believe in either case it isnt too complex to build and operate. I will stick to what you guys agree upon.
</blockquote></details>

I suppose the vast majority of use cases involve wiring a plant to the visualizer so encapsulating it in a Diagram will be OK.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: 5 of 7 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 21 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYKrKsg4STIlYbaWSZh:-KYOesLw4lX38JpCUB6P:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f8bbd1036aa7f78ba2a9e458aaf19a649b83d720/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L21)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

I suppose the vast majority of use cases involve wiring a plant to the visualizer so encapsulating it in a Diagram will be OK.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

liangfok commented:
*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 19 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOpvMU4GZ08pJQM5Uz:-KYOO8wK8XWcEyhrnDvd:b-lmkmqd) ([raw file](https://github.com/robotlocomotion/drake/blob/ba35ce1e0ba1ddf03c3d4e031c61acd910d448d9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L19)):*
> ```objc
> // library.
> 
> /// A custom `Diagram` composed of a `RigidBodyPlant` and a `DrakeVisualizer`.
> ```

BTW, can you generate the Doxygen website snd verify that Diagram, RigidBodyPlant, and DrakeVisualizer show up as links? You might need to add namespaces to them.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Review status: 5 of 7 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOwm7YCiFWlWTHIE9Z:-KYOwm7Y0LCvmekY0311:b-rgbo5b) (no related file):*
There's some cpplint to fix

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Reviewed 2 of 5 files at r4.
Review status: 6 of 7 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Reviewed 1 of 5 files at r4.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: 3 of 7 files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOwm7YCiFWlWTHIE9Z:-KYV7keBGORkHhQDXi8A:bt2vr0y) (no related file):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

There's some cpplint to fix
</blockquote></details>

Fixed!

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_world_demo.cc, line 70 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOoCRL4VDr1YoPlKKr:-KYTF54FoGyWLv7uBglh:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ba35ce1e0ba1ddf03c3d4e031c61acd910d448d9/drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_world_demo.cc#L70)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

btw consider `auto` here
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 39 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOofM40hLCUdM_80iS:-KYTFOyBCSsyOiRbFP3R:br582e2) ([raw file](https://github.com/robotlocomotion/drake/blob/ba35ce1e0ba1ddf03c3d4e031c61acd910d448d9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L39)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW I think the compiler could infer the template type here.
</blockquote></details>

It struggled :( Didnt change.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc, line 72 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOoutA1Cy65P0A_OeR:-KYUBlqcldPuCB_xqxmg:b-ukva8k) ([raw file](https://github.com/robotlocomotion/drake/blob/ba35ce1e0ba1ddf03c3d4e031c61acd910d448d9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.cc#L72)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW ditto
</blockquote></details>

Ditto reply.. compiler couldnt pick it up. 

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 19 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOpvMU4GZ08pJQM5Uz:-KYV7hfhG6D7NONWPWAB:bduqdi8) ([raw file](https://github.com/robotlocomotion/drake/blob/ba35ce1e0ba1ddf03c3d4e031c61acd910d448d9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L19)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, can you generate the Doxygen website snd verify that Diagram, RigidBodyPlant, and DrakeVisualizer show up as links? You might need to add namespaces to them.
</blockquote></details>

Done. Yes they were needed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.cc, line 74 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOtulc1c46FLz5U1Vh:-KYU7yIiUoq4zBcIY-P8:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ba35ce1e0ba1ddf03c3d4e031c61acd910d448d9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.cc#L74)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

BTW I think you could use `spruce.hh` for this parsing
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h, line 99 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYOpgkHBJSHzx6jFEGE:-KYU8t-kg4TsoDP2DBF3:byxu758) ([raw file](https://github.com/robotlocomotion/drake/blob/ba35ce1e0ba1ddf03c3d4e031c61acd910d448d9/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_tree_builder.h#L99)):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

I'm a little confused about how a model value is a string.  Is it a filename?
</blockquote></details>

Yes its a file name. Edited the comment a bit to reflect that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
rebased.. can squash and merge when the reviewers are ok with this PR.

---

Review status: 3 of 7 files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYV87uV77qSgkMfbzrE:b2wkm1i)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Just one BTW comment.

---

Reviewed 1 of 5 files at r4, 4 of 5 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 43 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYVEzYdeIRKMbLU8Qcf:-KYVEzYeEcaejNA36egB:b-8il2xo) ([raw file](https://github.com/robotlocomotion/drake/blob/4a870beea25f549b1c05cac6b8d30ed17ce01b54/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L43)):*
> ```objc
> /// A custom `Diagram` consisting of a `ConstantVectorSource` of 0 magnitude
> /// connected to a `VisualizedPlant`'s input port zero. The resulting diagram
> /// has no input
> ```

BTW: nit: odd newline.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYVFFY_zJ_2NOIOXWE4:bgvm1qr)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Looks like you're good to merge.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-:-KYVvcb5n4og_k5oCLwV:bpor7gl)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h, line 43 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/4366#-KYVEzYdeIRKMbLU8Qcf:-KYYp0vlzJCFR2HqB_xo:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4a870beea25f549b1c05cac6b8d30ed17ce01b54/drake/examples/kuka_iiwa_arm/iiwa_world/world_sim_diagram_factory.h#L43)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: nit: odd newline.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4366)*
<!-- Sent from Reviewable.io -->
